commands/netobserv

```diff
@@ -72,7 +70,56 @@ function packets() {
   esac
 }
 
-case "$1" in  
+function compare_versions() {
+    IFS="." read -r -a ver1 <<< "$1"
+    IFS="." read -r -a ver2 <<< "$2"
+
+    for ((i = 0; i < ${#ver1[@]} || i < ${#ver2[@]}; i++)); do
+        v1=${ver1[i]:-0} # Default to 0 if unset
+        v2=${ver2[i]:-0}
+        if ((v1 < v2)); then
+            result=0 # less than
+            return
+        elif ((v1 > v2)); then
+            result=1 # greater than
+            return
+        fi
+    done
+    result=2 # equal
+}
+
+# Required version of yq
+REQUIRED_YQ_VERSION="v4.43.1"
+YQ_BIN=$(which yq)
+# Get the current version of yq
+CURRENT_YQ_VERSION=$("$YQ_BIN" --version | awk '{print $NF}')
+compare_versions "${CURRENT_YQ_VERSION#v}" "${REQUIRED_YQ_VERSION#v}"
+# Check if the version matches and install it if needed
+if [ "$result" -eq 0 ]; then
+    echo "Installing yq version $REQUIRED_YQ_VERSION is required. Found $CURRENT_YQ_VERSION."
+    OS=$(uname | tr '[:upper:]' '[:lower:]') # linux or darwin
+    ARCH=$(uname -m)
+    if [ "$ARCH" == "x86_64" ]; then
+      ARCH="amd64"
+    elif [[ "$ARCH" == aarch64* ]]; then
+      ARCH="arm64"
+    elif [[ "$ARCH" == ppc64* ]]; then
+      ARCH="ppc64le"
+    elif [[ "$ARCH" == s390* ]]; then
+      ARCH="s390x"
+    fi
```

Comment:
As same as version, `ARCH` should come from arguments. 
Check `MULTIARCH_TARGETS` in the Makefile.

For now, we only manage `amd64` but we will pass the targeted archs in the dockerfiles for downstream builds in the future.

Reply:
fixed